vivisect/analysis/elf/elfplt_late.py

```diff
@@ -0,0 +1,376 @@
+'''
+Late-PLT analysis.
+
+During codeflow analysis, we detect PLT entries organically (and consequently 
+*accurately*), but often not all of them.  This analysis module applies a few
+different algorithms to identify the ones which were not discovered during 
+autoanalysis.
+
+Formerly, we attempted to do this analysis before all other code analysis
+but due to so many varieties of PLT creation by different compilers on 
+numerous hardware platforms and OSes, that method lead to many errors in PLT-
+identification, often hampering vital parts of vulnerability-research tools.
+We finally pulled the plug on preemptive analysis, in favor of letting auto-
+analysis identify "truth" and then filling in the gaps.
+'''
+
+
+import logging
+import vivisect
+import envi.common as e_cmn
+
+from . import elfplt
+
+logger = logging.getLogger(__name__)
+
+def analyze(vw):
+    logger.info('ELFPLT late-analysis')
+
+    for pltva, pltsz in elfplt.getPLTs(vw):
+        try:
+            analyzePLT(vw, pltva, pltsz)
+
+        except Exception as e:
+            logger.warning("Error in PLT-late analysis: %r", e, exc_info=True)
+
+
+def analyzePLT(vw, pltva, pltsz):
+    '''
+    Analyze a specific PLT section (there are often more than one and they often
+    different in format)
+
+    We make use of two different algorithms for each.
+    The first algorithm measures the distance from the start of known good PLT
+    functions and the GOT-referencing branch.  The only weakness is that this
+    method requires that the xref to the GOT is already identified (without 
+    throwing and emulator in there).  
+
+    The second algorithm measures the distance between known good PLT entries
+    and then attempts to identify divisors (up to 16 splits) which would make
+    more than one PLT function fit between them.  These attempted splits are 
+    validated using heuristics of the potential functions which would be created
+    by the division.  PLT entries in the same PLT section are incredibly similar
+    (not including the LazyLoader sometimes found at the beginning of a PLT)
+    '''
+    logger.info("PLT Section:  0x%x:%d", pltva, pltsz)
```

Comment:
Since this might be buried underneath a lot of other things, could you reformat the message to be more explicit about what it's logging, like:

`logger.info("PLT: Address: 0x%x, Size: %d", pltva, pltsz)`

Reply:
sure!